BuildConfigGen/Program.cs

```diff
@@ -375,6 +376,20 @@ private static void WriteNodePackageJson(string taskOutputNode, string nodeVersi
             ensureUpdateModeVerifier!.WriteAllText(outputNodePackagePath, nodePackageContent, suppressValidationErrorIfTargetPathDoesntExist: false);
         }
 
+        private static void WriteDevDependencyPackageJson(string taskOutputNode, string nodeVersion)
+        {
+            string outputNodePackagePath = Path.Combine(taskOutputNode, "package.json");
+            JsonNode outputNodePackagePathJsonNode = JsonNode.Parse(ensureUpdateModeVerifier!.FileReadAllText(outputNodePackagePath))!;
+            
+            // Upgrade typescript version for Node 20
+            if (nodeVersion == "^20.3.1") {
```

Comment:
Looks like we can just add this to WriteNodePackageJson rather than adding a new function

Reply:
Yes looks that it's redundant. I removed this one and moved this statement to the old function.